### DISCUSSION TITLE
my download speed seems capped at 130 kbps?

### DISCUSSION THREAD

**Discussion by tombr77** [aclap-dev/video-downloadhelper#1275] [Help and Q&A]
### Your question

i can  have several concurrent streams, but each has same limit.  

### Addon & CoApp version and details

version: 9.1.0.44
channel: stable
build date: 2024-09-04
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): b4578a54b4324edf
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:130.0) Gecko/20100101 Firefox/130.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\tom",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

What you describe sounds to me like throttling imposed by your serving web site.  If you're dealing with web sites that do not require logins, do please tell us what they are.  I must say that I have not seen anything like what you're describing.   I usually get at least 5 million bytes per second on YouTube.  There's a subscription site I download from that routinely gives over 30 million bytes per second, even over 60 million bytes per second on good days.  Medici TV usually gives between 2 & 10 million bytes per second for archived material.  Livestreams usually are in the 600,000 to 800,000 bytes per second range.  That is all that is needed to transmit 1920x1080 content.  My local CBS affiliate usually gives a bit over 1 million bytes per second for their NFL livestreams.  They're a bit higher because they are not just 1920x1080 but also 60 frames per second & particularly high bit rates.  The picture is much higher quality than your run-of-the-mill YouTube junk.  So I have to say my experience is much different from yours.  But it would help if you could share some URLs.  Not just web site names.  Particular URLs.  Of course, they must not require logins.  If they do, there's other things we can try.

**Comment by tombr77** [comment]
here are two i just used VDH on
https://cancerdecodedseries.com/episode-8-live/
https://healingkitchen.thesacredscience.com/episode7fjv/
and one from youtube
https://www.youtube.com/watch?v=C549o6jrqEw

I wish my internet was as fast as yours!  DSL with 3 mbps cap had been my
only option until recently.  i switched to a wireless hotspot that provides
30 mbps -- ( verified with ookla speedtest.  the Max Download was about 130
kbps with DSL --- i expected it to go higher with the new wireless hotspot,
but it remained the same....which i find puzzling, as i had assumed the
limit was imposed by the DSL....

tom

**Comment by tombr77** [comment]
another thing --- i choose 'quick audio only' download,  as the video uses
BW and doesnt usually provide extra information -so i am basically
converting these to 'podcasts'....

**Comment by tombr77** [comment]
( that 130 kbps limit seems to be across all things i've downloaded for a
long time - i have a habit of looking at the rate and 'time to completion'
after initiating a download .  While i know a specific web server could
cause an issue, i doubt that is it)

**Comment by GrampaWildWilly** [comment]
Get this out of the way.  My setup is Windows 11 Pro 64-bit version 23H2 build 22631.4169, Firefox 130.0 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20.

I had DSL for years.  Cincinnati Bell put DSL service into my part of the city the last of all the suburbs.  I thought it was wonderful, compared to a land line with a 56K dial-up modem that never actually went faster than 26.6K.  Their excuse being the phone lines around here were too old to actually do 56K.  But they finally condescended to make DSL available around here.  That used to top out around 550,000 bytes per second.  Bytes.  You didn't specify whether your speed rating was in bits or bytes.  Service providers like to give their ratings in bits because that multiplies the number by 8.  But nobody thinks in bits.  What is your speed rating in bytes?

I waited years for the phone company to put in their fiber optic service around here.  I finally got fed up waiting about 3 & 1/2 years ago.  I ditched their phone & Internet services.  I then added Internet & cell phone services to my existing cable TV subscription.  Now I'm getting what they advertise as 500Mbps.  But that's bits.  Divide by 8.  It's nominally 62.5 million bytes per second.  Bytes.  By the way, the phone company still hasn't installed fiber optic here.  I'm not out in the boonies.  I'm just in the westernmost region of the county, but still within city limits.  They are wankers.

So I don't want you to think I'm rubbing it in with what I'm about to post.  My purpose here is not to boast about my great Internet service.  There are services that are even better than mine.  But I have no complaints about what I have.  I do sympathize with your situation.  I've been there.

So let's get to your first example.

![#01](https://github.com/user-attachments/assets/c6a09ff7-829a-438a-81e0-4c7c05623661)
![#02](https://github.com/user-attachments/assets/913e943e-69e8-4cef-833f-f44a6fe5dd5e)
![#03](https://github.com/user-attachments/assets/84c09d63-2a31-4668-bb23-57ab9f0accb6)

That's 2,913,272,821 bytes downloaded in 221 seconds for an average download speed of a little over 13 million bytes per second.

**Reply by GrampaWildWilly** [reply]
![#04](https://github.com/user-attachments/assets/6f160fca-0ea3-4059-ab52-8088f8f442dd)
![#05](https://github.com/user-attachments/assets/4efcb0f1-eb94-45f3-8617-77dbe3212b4f)
![#06](https://github.com/user-attachments/assets/7b7f9bea-4505-4f3f-9276-4b334eaa0ccb)

Here it's 2,869,995,517 bytes downloaded in 117 seconds for an average download speed of about 24.5 million bytes per second.

**Reply by GrampaWildWilly** [reply]
![#07](https://github.com/user-attachments/assets/392d5071-1787-4275-9206-57e71be9e40e)
![#08](https://github.com/user-attachments/assets/700e41f2-6df0-439d-8a7c-a12e29eed613)
![#09](https://github.com/user-attachments/assets/33ab0e85-5326-417e-be0e-407c7fb86576)

Here it's 208,481,548 bytes downloaded in 14 seconds for an average download speed of a bit under 15 million bytes per second.  YouTube was feeling magnanimous this evening.

**Reply by GrampaWildWilly** [reply]
These examples show that the various sites are not throttling their services.  So the problem has to be at your end.  Is it in your computer?  Or is it just a slow ISP?  Let's hear from you what your ISP is telling you that you are buying from them.  In bytes per second.  Bytes.  Not what they advertise.  Bytes.

**Reply by GrampaWildWilly** [reply]
Oh.  Wait.  You already said it was 30Mbps.  Assuming that's bits, your nominal speed is 30/8 = something around 3.75 million bytes per second.  That is quite an adequate speed.  You should not be trying to conserve bandwidth by downloading audio-only content.  You should not be getting such putrid download speeds as 130,000 bytes per second.  You should be getting more than 25 times that speed.  Something is definitely wrong.  But I think my tests here eliminate VDH as the culprit.  I think you have a hardware issue.  I'd probably complain to my ISP if I were in your shoes.  They will no doubt try to blame your computer setup.  Challenge them to prove that something is wrong in your computer.  Make them bring one of their own computers (probably a laptop) to your home & hook it up to your connection & then see what their computer gets.  Stand up for your rights.

**Reply by GrampaWildWilly** [reply]
In my mind I'm rehearsing your conversation with your service provider.  I'm paying you jokers for a level of service & I'm not getting it.  Send one of your technicians out here with a laptop computer.  Hook it up to my connection & let's see what kind of speed you get.  I'm paying you but I'm not getting value for my money.  This is your problem.  Get out here & fix it.  Quit trying to make me feel like an idiot.  Instead of customer abuse, put your effort into customer service.  Fix your problem.  You do have a problem.  Fix it.

**Comment by tombr77** [comment]
thanks for looking into this.;.  and yes, the bits/bytes thing always trips
people up.
I assumed it was limited by my DSL, , which was 3 Mbps limit, and @ 130
kbps * 8 - about 1 Mbps - so with three running , each at 130, seems to
match.  Didnt explain why when i downloaded a single stream it was still
only 130 kbps.
when i switched to cellular wireless internet - and typical 25-33 Mbps, i
expcted faster., but it didnt change.   So perhaps it is my PC.   i will
try with (newer ) laptop and see....maybe it is my PC,  but not sure
why/how the desktop would create this limit....

tom

ps  i do audio only as most of the things are just people talking, and i
move the files to mp3 to put on my phone as its easier to carry around than
the desktop computer   wink wink.

the other problem i've been having is VDH not recognizing the content ,
even when playing.-- various sites -- reload the page several times and
sometimes it does see it..  this one seems to be getting worse.

**Comment by tombr77** [comment]
laptop experience similar speeds; however with one stream it drifted up to
160 kBps,  but dropped back to 130k each with two streams, and then slower
yet with three - about 270 K total.

i think i will open task manager and look at processes / CPU, memory - to
see if something in PC hits limit...  Its a little tricky when VDH often
doesnt recognize content....

tom

**Comment by tombr77** [comment]
hope my replies are reaching you... i dont see GrandpawildWilly in 'to'
address field...

this is weird -- if i do a quick download instead of 'audio only', my speed
gets higher -- up to 450 kBps...at mid resolution,  and 1.3 MBps at highest
resolution....
I wonder if  VDH under the hood is actually downloading the whole video,
then extracting the audio, rather than doing an ' only audio download' as
the description implies....
tom

**Reply by GrampaWildWilly** [reply]
You should not see my E-mail address in your destination address field.  I do not want E-mails of posts echoing what I can read on the web site.  Some people like to interact here via E-mail.  I do not.  I just get on GitHub in Firefox when it's convenient for me.  So I'm glad you're not sending me any E-mails that just duplicate what I can -- and do -- read here on the web site.

Like I said, I'm just another user.  So I have no idea what is going on under the covers to extract the audio from the video.  I would hope that the CoApp is using ffmpeg by supplying a `-map` parameter to select just the audio track.  I would hope the CoApp is not actually downloading the whole video onto your system & then extracting the audio track from that.

Interesting what you say about a speedup when you do a "quick download."  I'm suspicious of that terminology.  That is antique terminology from past releases of VDH.  The current terminology is `Download` & `Download As`.  What exactly are you doing?

When you are monitoring things, you should also use the Resource Monitor.  It shows you network activity on the Network tab, as I show above in some of my attached images.  It gives line usage in bytes per second.  I would like to keep this discussion in the bytes per second realm.  It is the way people think.  I hope you are giving your download speeds here in bytes per second.  You are somewhat confusing me with how you are describing things.  I'm not quite sure whether you are quoting bits or bytes.  Please always quote bytes per second.

The fact that you can get things to speed up when you just try to download a whole video tells me that there is some extra processing going 0n behind the scenes.  There is the ffmpeg parameter `-codec: copy` that the CoApp should be using.  It could be that under the covers, when you ask for an audio-only download, the CoApp has left out that parameter.  The absence of that parameter causes ffmpeg to perform a great deal of extra work.  That might manifest itself in slower download speeds.  I've experienced some of that at times.  For an audio-only download, in addition to `-map`, the CoApp should also be passing `-codec: copy` to ffmpeg.  If it is not, I would consider it a bug.  But this is all speculation.  Only @paulrouget Paul, the developer, could answer us on these questions.

If you can get better download speed by doing a `Download As` of the whole video instead of just the audio, you should probably do that.  Then use ffmpeg yourself to extract the audio track.  You'll need to learn a bit about ffmpeg to do this.  It's not hard, just that you might have to learn some new things.

I have a livestream I recorded the other day off Medici TV.  I extracted its audio track like this:

"C:\DownloadHelper CoApp\ffmpeg" -i "M:\Livestreams\Beethoven Tchaikovsky Rachlin Bronfman ChOrchEur 20240914 Eisenstadt.mp4" -map 0:a -codec: copy "M:\VDH Testing\Extracted Audio Track.mp4" 2>"M:\VDH Testing\Extracted Audio Track Log.txt"

Let me simplify this:

"path\ffmpeg" -i "recorded livestream file" -map 0:a -codec: copy "extracted audio track" 2>"log file"

You need to specify complete file specifications for the files names.  You have ffmpeg on your system in whatever directory you installed the CoApp.  Ffmpeg comes with the CoApp.  You can see where I installed mine.

The `-i` parameter specifies the video file on your system whose audio track you wish to extract.  You need to provide the name of whatever file you want to process.

The `0` part of `-map 0:a` tells ffmpeg to look in input file 0.  The input files are numbered sequentially starting at 0.  Since there is only one input, it is number 0.  `:a` tells ffmpeg to extract the audio track.  There can be multiple tracks in a file.  In a video, there are at least a video track & an audio track.  If you don't code a `-map` parameter, ffmpeg copies all the tracks in the input file.  So this form of `-map` selects just one track.

`-codec: copy` tells ffmpeg to just copy the input without doing any processing on it.  Ffmpeg can do a lot of different things to input, most of which I don't know because I've never needed to do anything fancy with it.  This parameter tells ffmpeg to do the simplest data copy.

The output file specification is what follows.  You need to provide whatever name you are going to use.

`2>` at the end is standard Windows redirection.  It captures the output of ffmpeg in a log file.  Here's this one:

[Extracted Audio Track Log.txt](https://github.com/user-attachments/files/17037063/Extracted.Audio.Track.Log.txt)

At the beginning of that log you can see full information about the input file.  It shows the video track & the audio track.  Those are the only tracks in this video.

Here's what the extracted audio track looks like:

![#01](https://github.com/user-attachments/assets/e4f6b02d-5112-4f69-8061-801a53eaeb16)

Doing this could be a lot faster than trying to download just the audio of something.  Try it & see.

**Reply by GrampaWildWilly** [reply]
Just as an aside, the audio track of 147M was extracted from a video whose size is 4.83G.  That just emphasizes how little data is needed for audio information compared to video information.
